src/Workspaces/Core/Portable/NamingStyles/NamingStyleRules.cs

```diff
@@ -46,5 +62,33 @@ private bool IsSymbolNameAnalyzable(ISymbol symbol)
 
             return true;
         }
+
+        private bool IsInterfaceImplementation(ISymbol symbol)
+        {
+            var containingType = symbol.ContainingType;
+            if (containingType == null)
+            {
+                return false;
+            }
+
+            var implementedTypes = containingType.AllInterfaces.Concat(containingType.GetBaseTypes().ToImmutableArray());
```

Comment:
Don't need to do anything if the symbol is non-public, or is static, or is itself in an interface.